src/lib/node/message_handler.ts

```diff
@@ -73,9 +79,14 @@ export class NodeMessageHandler {
           actualFirmware.data,
```

Comment:
Shouldn't you raise an error here before trying to begin a firmware update if one is in progress?

Reply:
We're only keeping track of whether the firmware update is in progress in between when the firmware gets uploaded to the controller and when the first event fires. Should I just make it a generic in progress status and only set it back to false once we get the `firmware update complete` event? Then we can catch the command earlier and throw the error. BTW the controller does throw an in progress error, but I suppose it's more efficient to catch it here